problems/[230124] 31. Next Permutation.js

```diff
@@ -0,0 +1,27 @@
+/**
+ * leetcode problem link: undefined
+ *
+ * @param {number[]} nums
+ * @return {void} Do not return anything, modify nums in-place instead.
+ */
+const nextPermutation = function (nums) {
+  const result = [];
+
+  const dfs = (i, nums) => {
+    if (i === nums.length) {
+      return result.push([...nums]);
+    }
+
+    for (let j = i; j < nums.length; j++) {
+      [nums[i], nums[j]] = [nums[j], nums[i]];
+
+      dfs(i + 1, nums);
+
+      [nums[i], nums[j]] = [nums[j], nums[i]];
+    }
+  };
+
+  dfs(0, nums);
```

Comment:
```suggestion
const dfs = (i, nums) => {
  const result = [];

  if (i === nums.length) {
    result.push([...nums]);
    return result;
  }

  for (let j = i; j < nums.length; j++) {
    [nums[i], nums[j]] = [nums[j], nums[i]];

    result.push(...dfs(i + 1, nums));

    [nums[i], nums[j]] = [nums[j], nums[i]];
  }

  return result;
};

const nextPermutation = function (nums) {
  dfs(0, nums);
```

이렇게 `dfs` 내에 `result`가 있으면 따로 함수를 빼줄 수도 있지 않을까요?